src/App.js

```diff
@@ -34,8 +34,15 @@ export default function App({
     return createElement(PageContext.Provider, { value: page.props }, null)
   }
 
-  const renderChildren = children
-    || (({ Component, props, key }) => createElement(Component, { key, ...props }))
+  const renderChildren = children || (({ Component, props, key }) => {
+    const child = createElement(Component, { key, ...props })
+
+    if (Component.layout) {
```

Comment:
Since `layout` is supposed to contain a component, I think `Layout` looks more natural.

```suggestion
    if (Component.Layout) {
```

Reply:
Right, but if it's not a component (see below), but rather a callback, it's probably better to keep lowercase (like `Page.getInitialProps` in Next.js).